### DISCUSSION TITLE
First start issue on iOS

### DISCUSSION THREAD

**Discussion by e808080** [TobiasBuchholz/Plugin.Firebase#474] [General]
We're having an issue that the very first time we start our app the notifications aren't working. We have verified that the FCM token should be correct (sending a push to the token gets a result and not "requested entity not found" so the token should be valid. However, no push notification appears in the app. Restarting the app solves everything and push notifications works (for the same token).

Is this a known issue and is there something we can do to fix this? We are speculating that the timing or something could be wrong, the plugin code runs before the user has actually clicked Allow for allowing push notifications. Maybe there is some function we can run to reinitialize once we know the app has fully started and user has clicked allow? For Android it works very first time correctly.

**Comment by khalil2099** [comment]
may i know which ios version are you testing ? i cant get notification on ios 17, but i can get on ios 15.
